test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -6343,21 +6343,21 @@ BOOST_AUTO_TEST_CASE(function_types_sig)
 	CHECK_SUCCESS_NO_WARNINGS(text);
 	text = R"(
 		contract C {
-			function h() external {
+			function h() pure external {
 			}
-			function f() external returns (bytes4) {
-				function () external g = this.h;
+			function f() view external returns (bytes4) {
+				function () pure external g = this.h;
 				return g.selector;
 			}
 		}
 	)";
 	CHECK_SUCCESS_NO_WARNINGS(text);
 	text = R"(
 		contract C {
-			function h() external {
+			function h() pure external {
 			}
-			function f() external returns (bytes4) {
-				function () external g = this.h;
+			function f() view external returns (bytes4) {
+				function () pure external g = this.h;
```

Comment:
Hm, that's interesting. Will this be parsed correctly?

Reply:
And yes it is! :-)

Reply:
I was wondering too but it worked, since it is parsed by the same helper.

Reply:
This is the error otherwise:

```
sel2.sol:5:33: Error: Type function () pure external is not implicitly convertible to expected type function () external.
                                function () external g = this.h;
                                ^-----------------------------^
```

Reply:
Ah, interesting. We should actually allow that conversion at some point in the future.